src/main/java/io/cdap/plugin/dlp/SensitiveRecordFilter.java

```diff
@@ -0,0 +1,310 @@
+import io.cdap.cdap.etl.api.MultiOutputEmitter;
+import io.cdap.cdap.etl.api.MultiOutputPipelineConfigurer;
+import io.cdap.cdap.etl.api.MultiOutputStageConfigurer;
+import io.cdap.cdap.etl.api.SplitterTransform;
+import io.cdap.cdap.etl.api.StageMetrics;
+import io.cdap.cdap.etl.api.StageSubmitterContext;
+import io.cdap.cdap.etl.api.TransformContext;
+import io.cdap.cdap.format.StructuredRecordStringConverter;
+import io.cdap.plugin.gcp.common.GCPConfig;
+import io.cdap.plugin.gcp.common.GCPUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * This class <code>SensitiveRecordFilter</code> provides an easy way to filter sensitive PII data from stream. The
+ * class utilizes Data Loss Prevention APIs for identifying sensitive data. Depending on the filter confidence set by
+ * user, the class either sends input record on sensitive port or non-sensitive port.
+ *
+ * <p>
+ * In case of issue with invoking DLP, the plugin depending on user choice either chooses to skip record, error pipeline
+ * or send record to error port.
+ * </p>
+ */
+@Plugin(type = SplitterTransform.PLUGIN_TYPE)
+@Name(SensitiveRecordFilter.NAME)
+@Description(SensitiveRecordFilter.DESCRIPTION)
+public final class SensitiveRecordFilter extends SplitterTransform<StructuredRecord, StructuredRecord> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SensitiveRecordFilter.class);
+  public static final String NAME = "SensitiveRecordFilter";
+  public static final String DESCRIPTION = "Filters input records based that are sensitive.";
+  private static final String SENSITIVE_PORT = "Sensitive";
+  private static final String NON_SENSITIVE_PORT = "Non-Sensitive";
+
+  // Stores the configuration passed to this class from user.
+  private final Config config;
+  private StageMetrics metrics;
+
+  // DLP service client for managing interactions with DLP service.
+  private DlpServiceClient client;
+
+  public SensitiveRecordFilter(Config config) {
+    this.config = config;
+  }
+
+  /**
+   * Invoked during deployment of pipeline to validate configuration of the pipeline. This method checks if the input
+   * specified is 'field' type and if it is, then checks if the field specified is present in the input schema.
+   *
+   * @param configurer a <code>MultiOutputPipelineConfigurer</code> for configuring pipeline.
+   * @throws IllegalArgumentException if there any issues with configuration of the plugin.
+   */
+  @Override
+  public void configurePipeline(MultiOutputPipelineConfigurer configurer) {
+    super.configurePipeline(configurer);
+
+    MultiOutputStageConfigurer stageConfigurer = configurer.getMultiOutputStageConfigurer();
+    Schema inputSchema = stageConfigurer.getInputSchema();
+
+    try {
+
+      client = DlpServiceClient.create(getSettings());
+    } catch (IOException e) {
+      LOG.error(e.printStackTrace());
```

Comment:
We should include error message

Reply:
also why isnt it being collected by failure collector?

Reply:
Added failure collector